### ISSUE TITLE
Promise.all structures produce unlinked spans

### ISSUE THREAD

**Issue by wdalmut** [wdalmut/opentelemetry-plugin-mongoose#16]
If i use a structure like

```js
Promise.all([
  ModelName
  .find(where)
  .skip(0)
  .limit(25)
  .sort({ fieldName: 'asc' }),
  ModelName.countDocuments(where)
])
```

Spans are correctly generated but not linked with main root span (appears as two separated spans).

**Comment by wdalmut** [comment]
this issue is strange... Two spans are correctly generated but sadly with two completely different traceId

**Comment by wdalmut** [comment]
The problem seems related to `Promise.all` structure and promise concatenation

```js
Product.find().then(...) // everything looks good

Promise.find().then(() => Product.countDocuments()).then(...) // 1 span only correctly linked with the root span 

Promise.all([Product.find(), Product.countDocuments()).then(...) // no span linked with root span
```

**Comment by wdalmut** [comment]
I found the same problem with the opentelemetry mysql-plugin https://github.com/open-telemetry/opentelemetry-js/issues/940

**Comment by wdalmut** [comment]
This should be fixed by #34
